src/core/MOM_forcing_type.F90

```diff
@@ -99,20 +99,22 @@ module MOM_forcing_type
 
   ! water mass fluxes into the ocean [R Z T-1 ~> kg m-2 s-1]; these fluxes impact the ocean mass
   real, pointer, dimension(:,:) :: &
-    evap        => NULL(), & !< (-1)*fresh water flux evaporated out of the ocean [R Z T-1 ~> kg m-2 s-1]
-    lprec       => NULL(), & !< precipitating liquid water into the ocean [R Z T-1 ~> kg m-2 s-1]
-    fprec       => NULL(), & !< precipitating frozen water into the ocean [R Z T-1 ~> kg m-2 s-1]
-    vprec       => NULL(), & !< virtual liquid precip associated w/ SSS restoring [R Z T-1 ~> kg m-2 s-1]
-    lrunoff     => NULL(), & !< liquid river runoff entering ocean [R Z T-1 ~> kg m-2 s-1]
-    frunoff     => NULL(), & !< frozen river runoff (calving) entering ocean [R Z T-1 ~> kg m-2 s-1]
-    seaice_melt => NULL()    !< snow/seaice melt (positive) or formation (negative) [R Z T-1 ~> kg m-2 s-1]
+    evap          => NULL(), & !< (-1)*fresh water flux evaporated out of the ocean [R Z T-1 ~> kg m-2 s-1]
+    lprec         => NULL(), & !< precipitating liquid water into the ocean [R Z T-1 ~> kg m-2 s-1]
+    fprec         => NULL(), & !< precipitating frozen water into the ocean [R Z T-1 ~> kg m-2 s-1]
+    vprec         => NULL(), & !< virtual liquid precip associated w/ SSS restoring [R Z T-1 ~> kg m-2 s-1]
+    lrunoff       => NULL(), & !< liquid river runoff entering ocean [R Z T-1 ~> kg m-2 s-1]
+    frunoff       => NULL(), & !< frozen river runoff (calving) entering ocean [R Z T-1 ~> kg m-2 s-1]
+    seaice_melt   => NULL()    !< snow/seaice melt (positive) or formation (negative) [R Z T-1 ~> kg m-2 s-1]
 
   ! Integrated water mass fluxes into the ocean, used for passive tracer sources [H ~> m or kg m-2]
   real, pointer, dimension(:,:) :: &
-    netMassIn   => NULL(), & !< Sum of water mass fluxes into the ocean integrated over a
-                             !! forcing timestep [H ~> m or kg m-2]
-    netMassOut  => NULL()    !< Net water mass flux out of the ocean integrated over a forcing timestep,
-                             !! with negative values for water leaving the ocean [H ~> m or kg m-2]
+    netMassIn     => NULL(), & !< Sum of water mass fluxes into the ocean integrated over a
+                               !! forcing timestep [H ~> m or kg m-2]
+    netMassOut    => NULL(), & !< Net water mass flux out of the ocean integrated over a forcing timestep,
+                               !! with negative values for water leaving the ocean [H ~> m or kg m-2]
+    netSalt       => NULL(), & !< Net salt entering the ocean [kgSalt m-2 s-1]
```

Comment:
> It looks netSalt got added to the type `forcing`, compared to `dev/ncar`. It looks like the GFDL->main PR removed it. So perhaps it slipped back in during conflict resolution when you merged the latest `dev/ncar` to your branch. Could you remove it?
> 
> Maybe also remove lines 3559-3560 of src/core/MOM_forcing_type.F90 that reference this type component in commented out code.

Good catch! Fixing that now.

Reply:
> I wonder if you should add
> 
> ```
>     call homogenize_field_t(fluxes%KPP_salt_flux, G)
> ```
> 
> at 3559-3560 of src/core/MOM_forcing_type.F90, to be consistent with the treatment of the other components of type `forcing`.

I'm guessing that you are right, but that would be opening up a can of worms. `KPP_salt_flux` is set with calls like

```fortran
    ! Sets: CS%KPP_buoy_flux, CS%KPP_temp_flux, CS%KPP_salt_flux
    ! NOTE: CS%KPP_buoy_flux, CS%KPP_temp_flux, CS%KPP_salt_flux are returned as rates (i.e. stuff per second)
    ! unlike other instances where the fluxes are integrated in time over a time-step.
    call calculateBuoyancyFlux2d(G, GV, US, fluxes, CS%optics, h, tv%T, tv%S, tv, &
                                 CS%KPP_buoy_flux, CS%KPP_temp_flux, CS%KPP_salt_flux)
```

So I think it would follow that we would want to homogenize `KPP_buoy_flux` and `KPP_temp_flux` as well, which we are not equipped to do. Also, from @alperaltuntas comments in https://github.com/NCAR/MOM6/pull/202#discussion_r818188426 we might be removing `KPP_salt_flux` from `forcing_type`.

> 
> I also wonder if there should be a call to `rotate_array` for `%KPP_salt_flux` in `rotate_forcing` in `src/core/MOM_forcing_type.F90`.

The more we pull on threads in the KPP module, the more I think it needs a pretty major overhaul...